### DISCUSSION TITLE
AWS S3 uploads to private buckets.

### DISCUSSION THREAD

**Discussion by mrtorks** [austintoddj/canvas#1269] [Troubleshooting] [answered]
This is in reference to #785 . I am using a private bucket that has restricted access and file uploads fail all the time. I am using the latest version of canvas. The code below internally throws the error. When I changed it form `storePublicly` to `store` it worked however the image preview does not show.
```
public function store()
    {
        $payload = request()->file();

        if (! $payload) {
            return response()->json(null, 400);
        }

        // Only grab the first element because single file uploads
        // are not supported at this time
        $file = reset($payload);

        $path = $file->storePublicly(Canvas::baseStoragePath(), [
            'disk' => config('canvas.storage_disk'),
        ]);

        return Storage::disk(config('canvas.storage_disk'))->url($path);
    }
 ```

**Comment by austintoddj** [comment]
What's the error when the image isn't showing?

**Reply by mrtorks** [reply]
There is no error. it just shows the standard broken image icon. I have attached screenshots.
<img width="2252" alt="Screen Shot 2023-05-23 at 9 30 49 AM" src="https://github.com/austintoddj/canvas/assets/13285039/57e98fe6-e83d-4de5-9eaf-3f363397392f">
<img width="1073" alt="Screen Shot 2023-05-23 at 9 30 57 AM" src="https://github.com/austintoddj/canvas/assets/13285039/09bb7837-14bd-4543-854e-1dc650ac1b04">

**Reply by austintoddj** [reply, marked as answer]
I see what happened. #785 correctly changed the store method, but I eventually refactored the MediaController to the UploadsController, and never carried the change over. That was my fault.

Going to work on a hotfix to get this updated in the `master` branch.

**Reply by austintoddj** [reply]
Added this to the latest release [6.0.46](https://github.com/austintoddj/canvas/releases/tag/v6.0.46)

**Comment by austintoddj** [comment]
Closing as obsolete with [Canvas v7.0.0](https://github.com/austintoddj/canvas/releases/tag/v7.0.0).
